src/pages/workspaces/workspace/WorkspaceTabs.ts

```diff
@@ -0,0 +1,96 @@
+import _ from 'lodash/fp';
+import { Dispatch, Fragment, ReactNode } from 'react';
+import { h } from 'react-hyperscript-helpers';
+import { analysisTabName } from 'src/analysis/runtime-common-components';
+import { TabBar } from 'src/components/tabBars';
+import * as Nav from 'src/libs/nav';
+import {
+  hasProtectedData,
+  isAzureWorkspace,
+  isGoogleWorkspace,
+  isOwner,
+  protectedDataMessage,
+  regionConstraintMessage,
+  WorkspaceWrapper as Workspace,
+} from 'src/libs/workspace-utils';
+import { WorkspaceAttributeNotice } from 'src/pages/workspaces/workspace/WorkspaceAttributeNotice';
+import { WorkspaceMenu } from 'src/pages/workspaces/workspace/WorkspaceMenu';
+
+export interface WorkspaceTabsProps {
+  namespace: string;
+  name: string;
+  workspace?: Workspace;
+  activeTab?: string;
+  refresh: () => void;
+  setDeletingWorkspace: Dispatch<boolean>;
+  setCloningWorkspace: Dispatch<boolean>;
+  setSharingWorkspace: Dispatch<boolean>;
+  setShowLockWorkspaceModal: Dispatch<boolean>;
+  setLeavingWorkspace: Dispatch<boolean>;
+}
+
+export const WorkspaceTabs = (props: WorkspaceTabsProps): ReactNode => {
+  const {
+    workspace,
+    activeTab,
+    refresh,
+    setDeletingWorkspace,
+    setCloningWorkspace,
+    setSharingWorkspace,
+    setShowLockWorkspaceModal,
+    setLeavingWorkspace,
+  } = props;
+  const { namespace, name } = props;
+  const wsOwner = !!workspace && isOwner(workspace.accessLevel);
+  const canShare = workspace?.canShare;
+  const isLocked = !!workspace?.workspace.isLocked;
+  const workspaceLoaded = !!workspace;
+  const googleWorkspace = workspaceLoaded && isGoogleWorkspace(workspace);
+  const azureWorkspace = workspaceLoaded && isAzureWorkspace(workspace);
+
+  const onClone = () => setCloningWorkspace(true);
+  const onDelete = () => setDeletingWorkspace(true);
+  const onLock = () => setShowLockWorkspaceModal(true);
+  const onShare = () => setSharingWorkspace(true);
+  const onLeave = () => setLeavingWorkspace(true);
+
+  const tabs = [
+    { name: 'dashboard', link: 'workspace-dashboard' },
+    { name: 'data', link: 'workspace-data' },
+    { name: 'analyses', link: analysisTabName },
+    ...(googleWorkspace
+      ? [
+          { name: 'workflows', link: 'workspace-workflows' },
+          { name: 'job history', link: 'workspace-job-history' },
+        ]
+      : []),
+    ...(azureWorkspace ? [{ name: 'workflows', link: 'workspace-workflows-app' }] : []),
+  ];
+  return h(Fragment, [
```

Comment:
This Fragment looks unnecessary.

Reply:
Wrong branch >_<
It makes a bit more sense in the container anyway.